azure/packages/azure-client/src/AzureClient.ts

```diff
@@ -228,25 +228,22 @@ export class AzureClient {
         );
 
         const rootDataObject = await requestFluidObject<RootDataObject>(container, "/");
-        return new (class extends FluidContainer {
-            /**
-             * See {@link FluidContainer.attach}
-             *
-             * @remarks This is required since the FluidContainer doesn't have knowledge of how the attach will happen
-             * or the id that will be returned.
-             * This exists because we are projecting a separation of server responsibility to the end developer that
-             * doesn't exist in the framework.
-             */
-            public async attach(): Promise<string> {
-                if (this.attachState !== AttachState.Detached) {
-                    throw new Error("Cannot attach container. Container is not in detached state");
-                }
-                await container.attach(createNewRequest);
-                const resolved = container.resolvedUrl;
-                ensureFluidResolvedUrl(resolved);
-                return resolved.id;
+
+        /**
+         * See {@link FluidContainer.attach}
+         */
+        const attach = async (): Promise<string> => {
+            if (container.attachState !== AttachState.Detached) {
+                throw new Error("Cannot attach container. Container is not in detached state");
             }
-        })(container, rootDataObject);
+            await container.attach(createNewRequest);
```

Comment:
Would here one option be to pass IRequest to attach, and let this logic reside in FluidContainer?

Reply:
I have lots of opinions about this area :-P

We kind of could.  However, there's a bigger problem here that attach never should have been on the Container or FluidContainer.  Attaching is a function of the service (i.e. should have been something like "service.publishContainer(container.serialize())").  Because of this inversion, we hit a lot of these issues of "how do I pass this thing into the container related to connection/attaching?".  This is part of why I've split out attach() as a separate callback returned by ModelLoader.createDetached() rather than putting it on the model.

I think we'll eventually want to move in that direction, but didn't feel I could make that change in main.

Reply:
Yeah, all of that would be a breaking change, so understood that you had few paths there. We can tackle this other topic separately, it would be interesting to see how you would reason around other APIs that imply relationship to service (ex. connected state/api-s etc)